crates/wdl-analysis/src/diagnostics.rs

```diff
@@ -405,7 +405,10 @@ pub fn no_common_type(
         "type mismatch: a type common to both type `{expected}` and type `{actual}` does not exist"
     ))
     .with_label(format!("this is type `{actual}`"), actual_span)
-    .with_label(format!("this is type `{expected}`"), expected_span)
+    .with_label(
+        format!("all elements up until and including this one had a common type `{expected}`"),
```

Comment:
this seems very wordy to me. A more compact option:
```suggestion
        format!("detected common type `{expected}` from this and prior elements"),
```
?